src/components/DirectionsOnMap/DirectionsOnMap.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Loader } from "@googlemaps/js-api-loader";
+import "./DirectionsOnMap.scss";
+import pokeballIcon from "../../assets/4.svg";
+import locationPointerIcon from "../../assets/location-pointer-blue.svg";
+
+const apiKey = import.meta.env.VITE_API_KEY;
+
+type PokemonLocations = {
+  latitude: number;
+  longitude: number;
+};
+
+const DirectionsOnMap: React.FC<PokemonLocations> = ({
+  latitude,
+  longitude,
+}) => {
+  const mapRef = useRef<HTMLDivElement>(null);
+  const [showDirections, setShowDirections] = useState(false);
+
+  useEffect(() => {
+    const fetchDirections = async () => {
+      try {
+        const loader = new Loader({
+          apiKey: apiKey,
+          version: "weekly",
+        });
+
+        const google = await loader.load();
+
+        const initializeMap = () => {
+          const mapOptions: google.maps.MapOptions = {
+            center: { lat: 32.06476, lng: 34.7716346 },
+            zoom: 17,
+            mapId: "myApp",
+          };
+
+          const map = new google.maps.Map(
+            mapRef.current as HTMLDivElement,
+            mapOptions
+          );
+
+          const addMarkersAndRoute = () => {
+            const originMarker = new google.maps.Marker({
+              position: new google.maps.LatLng(32.064584, 34.771829),
+              map,
```

Comment:
i would make a generic function for both Marker generating inside a utils file :-)
and then of course pass needed variables